src/compiler/checker.ts

```diff
@@ -26774,6 +26781,12 @@ namespace ts {
             diagnostics.add(resultDiagnostic);
         }
 
+        function containerSeemsToBeEmptyDomElement(containingType: Type) {
+            return (compilerOptions.lib && !compilerOptions.lib.includes("dom")) &&
+                everyContainedType(containingType, type => type.symbol && /^(EventTarget|Node|((HTML[a-zA-Z]*)?Element))$/.test(type.symbol.escapedName.toString())) &&

```

Comment:
escapedName is already a string, it just sometimes contains `__`. Casting it to string is OK here since you are comparing the name to a known list that doesn't contain leading underscores, or for Ultimate Correctness you can use the unescapeLeadingUnderscores functions.